lib/elixir/lib/range.ex

```diff
@@ -40,7 +40,8 @@ defmodule Range do
   defstruct first: nil, last: nil
 
   @type t :: %Range{}
-  @type t(first, last) :: %Range{first: first, last: last}
+  @type t(first :: integer, last :: integer) :: %Range{first: integer, last: integer}
```

Comment:
I don't think it's possible to write a typespec like that. In fact, the elixir compiler should reject `@type t(first :: integer, last :: integer) :: %Range{first: integer, last: integer}`. I will work on a fix.

The only reasonable solution is the one @josevalim suggested.